common/persistence/elasticsearch/esProcessor.go

```diff
@@ -233,8 +242,12 @@ func (p *esProcessorImpl) sendToAckChan(visibilityTaskKey string, ack bool) {
 		return
 	}
 
-	ackChWithStopwatch.addToAckStopwatch.Stop()
-	ackChWithStopwatch.ackCh <- ack
+	ackChWithStopwatch.addToProcessStopwatch.Stop()
+	// In case if sendToAckChan is called for already processed channel, just ignore it.
+	select {
+	case ackChWithStopwatch.ackCh <- ack:
+	default:
+	}
 
 	p.mapToAckChan.Remove(visibilityTaskKey)
```

Comment:
what if line number
240, then 245-250 is executed, then context switch, 168-180 is executed, then 252 
^ will break the contract